### PR TITLE
update mayor techno's utils

### PR DESCRIPTION
**Pull request by brunobrgGH**

### CODE REVIEW

File: src/main/kotlin/gg/skytils/skytilsmod/core/Config.kt [hunk 1]
```diff
@@ -1058,9 +1058,17 @@ object Config : Vigilant(
     @Property(
         type = PropertyType.SWITCH, name = "Show Shiny Orb Waypoints",
         description = "Creates a waypoint of where your shiny orbs are",
-        category = "Events", subcategory = "Technoblade"
+        category = "Events", subcategory = "Technoblade",
+        searchTags = ["Shiny", "Technoblade", "Pig"]
     )
     var shinyOrbWaypoints = false
+    @Property(
```
Comment (Sychic) [inline src/main/kotlin/gg/skytils/skytilsmod/core/Config.kt:1139]: Please add a newline between the last setting and this one.

File: src/main/kotlin/gg/skytils/skytilsmod/core/Config.kt [hunk 2]
```diff
@@ -1058,9 +1058,17 @@ object Config : Vigilant(
     @Property(
         type = PropertyType.SWITCH, name = "Show Shiny Orb Waypoints",
         description = "Creates a waypoint of where your shiny orbs are",
-        category = "Events", subcategory = "Technoblade"
+        category = "Events", subcategory = "Technoblade",
+        searchTags = ["Shiny", "Technoblade", "Pig"]
```
Comment (Sychic) [inline src/main/kotlin/gg/skytils/skytilsmod/core/Config.kt]: There's no need for the "shiny" or "technoblade" tags since they're already in the name and subcategory.

File: src/main/kotlin/gg/skytils/skytilsmod/core/Config.kt [hunk 3]
```diff
@@ -1058,9 +1058,17 @@ object Config : Vigilant(
...
     )
     var shinyOrbWaypoints = false
+    @Property(
+        type = PropertyType.SWITCH, name = "Show Shiny Pig Locations",
+        description = "Shows the location of the pig and draws a line to its orb.",
+        category = "Events", subcategory = "Technoblade",
+        searchTags = ["Shiny", "Technoblade", "Pig"]
```
Comment (Sychic) [inline src/main/kotlin/gg/skytils/skytilsmod/core/Config.kt]: Same as above.

File: src/main/kotlin/gg/skytils/skytilsmod/features/impl/events/TechnoMayor.kt [hunk 4]
```diff
@@ -20,62 +20,72 @@ package gg.skytils.skytilsmod.features.impl.events
...
-        if (e.customNameTag != "§6§lSHINY ORB") return
+        if (e !is EntityArmorStand || !e.hasCustomName() || e.isDead || e.customNameTag != "§6§lSHINY ORB") return
         val pos = e.position
         mc.theWorld.getEntitiesWithinAABBExcludingEntity(
             e,
             AxisAlignedBB(pos, BlockPos(pos.x + 1, pos.y + 1, pos.z + 1))
         ).find {
             if (it is EntityArmorStand && it.hasCustomName() && it.getCustomNameTag().contains(mc.thePlayer.name)) {
-                it.worldObj.removeEntity(it)
+                it.worldObj.removeEntity(it);
```
Comment (Sychic) [inline src/main/kotlin/gg/skytils/skytilsmod/features/impl/events/TechnoMayor.kt]: This semicolon is redundant.

File: src/main/kotlin/gg/skytils/skytilsmod/features/impl/events/TechnoMayor.kt [hunk 5]
```diff
@@ -20,62 +20,72 @@ package gg.skytils.skytilsmod.features.impl.events
...
         if (SBInfo.mode != SkyblockIsland.Hub.mode && SBInfo.mode != SkyblockIsland.FarmingIsland.mode) return
         if (!Skytils.config.shinyOrbWaypoints) return
 
+        val iterator = shinyPigs.entries.iterator();
+        while (iterator.hasNext()) {
+            val it = iterator.next();
+            if (it.value == null || it.value!!.isDead) {
+                iterator.remove()
+            }
+        }
```
Comment (Sychic) [inline src/main/kotlin/gg/skytils/skytilsmod/features/impl/events/TechnoMayor.kt]: This can be replaced with a `removeIf`
```suggestion
        shinyPigs.entries.removeIf { (_, pig) ->
                pig?.isDead != false
        }
```

File: src/main/kotlin/gg/skytils/skytilsmod/features/impl/events/TechnoMayor.kt [hunk 6]
```diff
@@ -86,32 +96,63 @@ object TechnoMayor {
...
-        if (packet.type.toInt() == 2) return
-        val unformatted = packet.chatComponent.unformattedText.stripControlCodes()
-        if (unformatted == "Your Shiny Orb and associated pig expired and disappeared." || unformatted == "SHINY! The orb is charged! Click on it for loot!") {
-            orbLocations.removeIf { pos: Vec3 ->
-                mc.thePlayer.position.distanceSq(
-                    pos.xCoord,
-                    pos.yCoord,
-                    pos.zCoord
-                ) < 7 * 7
+    @SubscribeEvent
```
Comment (Sychic) [inline src/main/kotlin/gg/skytils/skytilsmod/features/impl/events/TechnoMayor.kt:115]: newline

**Review by Sychic — CHANGES_REQUESTED** [review]

File: src/main/kotlin/gg/skytils/skytilsmod/features/impl/events/TechnoMayor.kt (hunk 5 above)
Comment (hannibal002) [reply src/main/kotlin/gg/skytils/skytilsmod/features/impl/events/TechnoMayor.kt]: `shinyPigs.values.removeIf { it?.isDead != false }` should work as well

File: src/main/kotlin/gg/skytils/skytilsmod/core/Config.kt [hunk 7]
```diff
@@ -3092,6 +3093,9 @@ object Config : Vigilant(
 
         addDependency("assumeWitherImpact", "witherShieldCooldown")
 
+        addDependency("showTrophyFishTotals", "trophyFishTracker")
+        addDependency("showTotalTrophyFish", "trophyFishTracker")
+
```
Comment (Sychic) [inline src/main/kotlin/gg/skytils/skytilsmod/core/Config.kt:3098]: Why is this here?

Comment (My-Name-Is-Jeff) [reply src/main/kotlin/gg/skytils/skytilsmod/core/Config.kt:3098]: image
??

Comment (Sychic) [reply src/main/kotlin/gg/skytils/skytilsmod/core/Config.kt:3098]: image

Comment (My-Name-Is-Jeff) [reply src/main/kotlin/gg/skytils/skytilsmod/core/Config.kt:3098]: ur thing is bugging ur comment aint even there

Comment (Sychic) [reply src/main/kotlin/gg/skytils/skytilsmod/core/Config.kt:3098]: I think it's because you used merge instead of rebase

Comment (My-Name-Is-Jeff) [reply src/main/kotlin/gg/skytils/skytilsmod/core/Config.kt:3098]: Is everything else fine?

Comment (Sychic) [reply src/main/kotlin/gg/skytils/skytilsmod/core/Config.kt:3098]: Yeah seems fine
